scripts/expo-upload-sourcemaps.js

```diff
@@ -101,6 +101,15 @@ function groupAssets(assetPaths) {
   return groups;
 }
 
+process.env.NODE_ENV = process.env.NODE_ENV || 'development';
```

Comment:
Does it have any side effects? Because if it doesn't I think we can leave it out.

Reply:
So `@expo/env` is a shim over the dotenv package, where falling back to 'development' would affect the precedence of `.env` files that are applied: https://github.com/bkeepers/dotenv?tab=readme-ov-file#customizing-rails

For most of the expo commands, we fallback to `development` unless there is a good reason to do something else (notable exceptions are defaulting to `production` for metro bundling purposes) . I can remove it though, I understand that this may add too much complexity for a simple script

Reply:
Thank you for pointing that out, let's keep the fallback and add a small comment about why.